src/cursor/cur_std.c

```diff
@@ -1196,59 +1179,48 @@ __wt_cursor_largest_key(WT_CURSOR *cursor)
 int
 __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
 {
+    WT_CONFIG_ITEM cval;
     WT_CURSOR_BTREE *cbt;
+    WT_DECL_CONF(WT_CURSOR, bound, conf);
     WT_DECL_RET;
     WT_ITEM key;
     WT_SESSION_IMPL *session;
     int exact;
-    char *cfg;
-    bool inclusive, have_action;
+    bool inclusive;
 
-    cfg = NULL;
     cbt = (WT_CURSOR_BTREE *)cursor;
     exact = 0;
-
-    /*
-     * Our API defines "inclusive" as true by default, it also defines "set" as the default action.
-     * This means we can't expect the user to have provided those configurations via the config
-     * string. Normally WiredTiger merges the user provided configuration with the default
-     * configuration, for performance reasons we are skipping this step. As such we need to handle
-     * the cases where the user did and didn't provide the config making up the difference for the
-     * defaults.
-     *
-     * These two booleans provide a mechanism to handle the user not providing the configuration and
-     * still being able to parse it.
-     */
     inclusive = true;
-    have_action = false;
 
     CURSOR_API_CALL(cursor, session, ret, bound, NULL);
+    WT_ERR(__wt_conf_compile_api_call(session, WT_CONFIG_REF(session, WT_CURSOR_bound),
+      WT_CONFIG_ENTRY_WT_CURSOR_bound, config, &_conf, sizeof(_conf), &conf));
 
     if (CUR2BT(cursor)->type == BTREE_COL_FIX)
         WT_ERR_MSG(session, EINVAL, "setting bounds is not compatible with fixed column store");
 
-    if (config == NULL || config[0] == '\0')
-        WT_ERR_MSG(session, EINVAL, "an empty config is not valid when setting or clearing bounds");
-
     /* Action is default to "set". */
-    if (CONFIG_VALUE_FOR(config, "action", cfg) != NULL)
-        have_action = true;
-
-    if (!have_action || WT_PREFIX_MATCH(cfg, "set")) {
+    WT_ERR(__wt_conf_gets(session, conf, action, &cval));
+    if (WT_CONF_STRING_MATCH(set, cval)) {
         if (WT_CURSOR_IS_POSITIONED(cbt))
             WT_ERR_MSG(session, EINVAL, "setting bounds on a positioned cursor is not allowed");
 
         /* The cursor must have a key set to place the lower or upper bound. */
         WT_ERR(__cursor_checkkey(cursor));
 
-        /* Inclusive is true by default. */
-        if (CONFIG_VALUE_FOR(config, "inclusive", cfg) != NULL && !WT_PREFIX_MATCH(cfg, "true"))
+        WT_ERR(__wt_conf_gets_def(session, conf, inclusive, true, &cval));
+        if (cval.val == 0)
             inclusive = false;
 
-        if (CONFIG_VALUE_FOR(config, "bound", cfg) == NULL)
+        WT_ERR(__wt_conf_gets(session, conf, bound, &cval));
+        /*
+         * FIXME-WT-12777 The above call shouldn't actually return a bound, but it does. We check
+         * the len to work around the bug mentioned in the ticket.
+         */
+        if (cval.len == 0)
             WT_ERR_MSG(session, EINVAL,
               "a bound must be specified when setting bounds, either \"lower\" or \"upper\"");
```

Comment:
@luke-pearson, I investigated this, and I believe it is already fixed in WT-12715.  if you merge develop into your PR branch, then I'm pretty sure you can get rid of the workaround code here.

Reply:
That worked, thanks Don!